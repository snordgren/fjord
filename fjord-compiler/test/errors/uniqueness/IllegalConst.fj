module errors.uniqueness.IllegalConst

illegalConst : String -* Int -> String
illegalConst x y =
  x

useIllegalConst : (Int -> String) -> (String, String)
useIllegalConst f =
  (f 0, f 0)

dup : String -* (String, String)
dup x = 
  useIllegalConst (illegalConst x)

myDup : (String, String)
myDup = 
  dup "hi"

