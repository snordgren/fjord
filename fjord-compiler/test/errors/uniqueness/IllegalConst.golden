test/errors/uniqueness/IllegalConst.fj:13:18:
   |
13 |   useIllegalConst (illegalConst x)
   |                  ^^^^^^^^^^^^^^^^^^
expression has type 1:(?:Int -> 1:String), expected ?:Int -> 1:String
